traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -1676,75 +1609,100 @@ func createV2(inf *api.APIInfo, w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
-	resultRows, err := inf.Tx.NamedQuery(insertQuery, server)
+
+	serverID, err := createServerV2(inf.Tx, server)
 	if err != nil {
 		userErr, sysErr, errCode := api.ParseDBError(err)
-		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
 		return
 	}
-	defer resultRows.Close()
 
-	rowsAffected := 0
-	for resultRows.Next() {
-		rowsAffected++
-		if err := resultRows.StructScan(&server.CommonServerProperties); err != nil {
-			api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("server create scanning: %v", err))
-			return
-		}
-	}
-	if rowsAffected == 0 {
-		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New("server create: no server was inserted, no id was returned"))
-		return
-	} else if rowsAffected > 1 {
-		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New("too many ids returned from server insert"))
+	ifaces, err := server.LegacyInterfaceDetails.ToInterfacesV4(*server.IPIsService, *server.IP6IsService, server.RouterHostName, server.RouterPortName)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
 		return
 	}
 
-	ifaces, err := server.LegacyInterfaceDetails.ToInterfacesV4(*server.IPIsService, *server.IP6IsService, server.RouterHostName, server.RouterPortName)
-	if err != nil {
-		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, err)
+	if userErr, sysErr, errCode := createInterfaces(int(serverID), ifaces, inf.Tx.Tx); userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
 		return
 	}
 
-	if userErr, sysErr, errCode := createInterfaces(*server.ID, ifaces, tx); err != nil {
-		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+	if server.UpdPending != nil {
+		if *server.UpdPending {
+			if err := dbhelpers.QueueUpdateForServer(inf.Tx.Tx, serverID); err != nil {
+				api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+				return
+			}
+		} else {
+			if err := dbhelpers.DequeueUpdateForServer(inf.Tx.Tx, serverID); err != nil {
+				api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+				return
+			}
+		}
+	}
+
+	if server.RevalPending != nil {
+		if *server.RevalPending {
+			if err := dbhelpers.QueueRevalForServer(inf.Tx.Tx, serverID); err != nil {
+				api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+				return
+			}
+		} else {
+			if err := dbhelpers.DequeueUpdateForServer(inf.Tx.Tx, serverID); err != nil {
+				api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+				return
+			}
+		}
+	}
```

Comment:
ditto an earlier comment: I don't think the "create/update server" APIs should deal with queuing/unqueuing updates/revals at all. There is another API for doing that already that should be the sole owner of this functionality IMO

Reply:
Ditto as well. I put various replies about this in other comments.

Should we remove the ability to set any config/reval updates via these server endpoints (PUT/POST/PATCH)?